src/datepicker/utils/date-utils-provider.js

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import DateFnsUtils from '@date-io/date-fns';
+
+const baseInstance = new DateFnsUtils();
+
+export const DateUtilsContext: React.Context<ThemeT> = React.createContext(
+  baseInstance,
+);
+
+const DateUtilsProvider = (props: {utils: any, children: ?React.Node}) => {
+  const {utils: Utils, children} = props;
+  const utilsInstance = React.useMemo(() => new Utils(), [Utils]);
```

Comment:
Initializing the `utils` class here feels a bit strange, can we guarantee the `date-io` adapters take no arguments? What's the downside to the provider's parent calling the constructor?

Reply:
Different types of the utils class from date-io can accept different arguments.  In the Material UI implementation, all of those arguments are just props on the DateUtilsProvider.  

I was pretty much just following the Material UI implementation across the board, but now that you mention it, just providing the instance as a prop is probably simpler.  I guess the only real benefit would be similarity to the API of Material UI's date swapping.

Reply:
I made this change, and it ended up simplifying the types as well